src/renderer/components/ProjectCreation/CloudConfigView.tsx

```diff
@@ -0,0 +1,168 @@
+
+  const handleAPIKeyInput = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const key = event.target.value;
+    setAPIKey(key);
+    setAwaitingAPIKey(key.length === 0);
+  };
+
+  const handleClientSecretInput = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const secret = event.target.value;
+    setClientSecret(secret);
+    setAwaitingClientSecret(secret.length === 0);
+  };
+
+  const saveButton = (
+    <PrimaryButton
+      onClick={saveCloudCredentials}
+      disabled={isAwaitingAPIKey || isAwaitingClientSecret}
+      fullWidth
+    >
+      Save
+    </PrimaryButton>
+  );
+
+  const cancelButton = (
+    <SecondaryButton onClick={prevView} fullWidth>
+      Back
+    </SecondaryButton>
+  );
+
+  const handleHelpClick: () => void = () => {
+    openExternalLink('https://mlvet.app');
+  };
+
+  const defaultText =
+    "This is your first time using the cloud transcription method. To get started you'll need to provide an API key and client secret for Google Cloud Speech-to-Text";
+
+  const text = textToDisplay ?? defaultText;
+
+  useKeypress(
+    saveCloudCredentials,
+    !(isAwaitingAPIKey || isAwaitingClientSecret),
+    ['Enter', 'NumpadEnter']
+  );
+
+  return (
+    <Container sx={{ height: { xs: 500 } }}>
+      <CustomRowStack justifyContent="space-between">
+        <Typography
+          overflow="hidden"
+          textOverflow="ellipsis"
+          variant="h1"
+          sx={{ color: colors.grey[400] }}
+        >
+          {projectName}
+        </Typography>
+        <IconButton
+          sx={{ color: colors.yellow[500], fontSize: 20 }}
+          onClick={closeModal}
+        >
+          <CloseIcon />
+        </IconButton>
+      </CustomRowStack>
+      <CustomColumnStack
+        justifyContent="space-between"
+        sx={{ height: '90.5%' }}
+      >
+        <CustomStack justifyContent="space-between">
+          <CustomStack>
+            <Typography variant="h3">
+              Cloud transcription configuration required
+            </Typography>
+            <Typography variant="p-400">{text}</Typography>
+            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+            <Link onClick={handleHelpClick}>
+              {/* Using the href prop of this component will break hence we have to open the URL in an external browser window */}
+              How can I get an API key and client secret? &gt;
+            </Link>
+          </CustomStack>
+        </CustomStack>
+        <CustomColumnStack>
+          <CustomStack>
+            <TextField
+              label="API Key"
+              value={APIKey}
+              onChange={(event) => handleAPIKeyInput(event)}
+              autoFocus
+            />
+          </CustomStack>
+          <CustomStack>
+            <TextField
+              label="Client Secret"
+              value={clientSecret}
+              onChange={(event) => handleClientSecretInput(event)}
```

Comment:
Since the event being passed in exactly matches your function signature you can make it point-free

`onChange={handleClientSecretInput}`

That said, a more common pattern is to extract the value from the event before passing it to the handler, since the handler isn't doing anything else with the event object. i.e.

`onChange={(event) => handleClientSecretInput(event.target.value)}`

is something you tend to see a lot in react form code

Obviously, same goes for the other one too

Reply:
Nice catch, I fixed it here: [57dc015](https://github.com/chloebrett/mlvet/pull/214/commits/57dc0156edd66b6fa1d27f5349dcf4314d0dc1d1)